File: src/main/java/com/yupi/airouter/service/BillingRecordService.java
package com.yupi.airouter.service;

import com.mybatisflex.core.paginate.Page;
import com.mybatisflex.core.service.IService;
import com.yupi.airouter.model.entity.BillingRecord;

import java.math.BigDecimal;
import java.util.List;

/**
 * 消费账单服务
 *
 * @author <a href="https://codefather.cn">编程导航学习圈</a>
 */
public interface BillingRecordService extends IService<BillingRecord> {

    /**
     * 获取用户的消费账单列表（分页）
     *
     * @param userId 用户ID
     * @param pageNum 页码
     * @param pageSize 每页大小
     * @return 分页结果
     */
    Page<BillingRecord> listUserBillingRecords(Long userId, int pageNum, int pageSize);

    /**
     * 获取用户的总消费金额
     *
     * @param userId 用户ID
     * @return 总消费金额
     */
    BigDecimal getUserTotalSpending(Long userId);

    /**
     * 获取用户的总充值金额
     *
     * @param userId 用户ID
     * @return 总充值金额
     */
    BigDecimal getUserTotalRecharge(Long userId);
}

File: go-backend/internal/service/billing_record_service.go
package service

import (
	"github.com/yupi/airouter/go-backend/internal/common"
	"github.com/yupi/airouter/go-backend/internal/errno"
	"github.com/yupi/airouter/go-backend/internal/model/entity"
	"github.com/yupi/airouter/go-backend/internal/repository"
)

type BillingRecordService struct {
	repo *repository.BillingRecordRepository
}

func NewBillingRecordService(repo *repository.BillingRecordRepository) *BillingRecordService {
	return &BillingRecordService{repo: repo}
}

func (s *BillingRecordService) ListUserBillingRecords(userID, pageNum, pageSize int64) (common.PageResponse[entity.BillingRecord], error) {
	if userID <= 0 {
		return common.PageResponse[entity.BillingRecord]{}, errno.New(errno.ParamsError)
	}
	if pageNum <= 0 {
		pageNum = 1
	}
	if pageSize <= 0 {
		pageSize = 10
	}
	return s.repo.ListByUserID(userID, pageNum, pageSize)
}

func (s *BillingRecordService) GetUserTotalSpending(userID int64) (float64, error) {
	if userID <= 0 {
		return 0, errno.New(errno.ParamsError)
	}
	return s.repo.SumAmountByType(userID, "api_call")
}

func (s *BillingRecordService) GetUserTotalRecharge(userID int64) (float64, error) {
	if userID <= 0 {
		return 0, errno.New(errno.ParamsError)
	}
	return s.repo.SumAmountByType(userID, "recharge")
}

func (s *BillingRecordService) CreateRecord(record *entity.BillingRecord) error {
	return s.repo.Create(record)
}
